File: backend/app/schemas/wallet.py
from datetime import datetime

from pydantic import BaseModel


class WalletOut(BaseModel):
    balance_credits: int


class WalletLedgerEntryOut(BaseModel):
    id: int
    delta_credits: int
    reason: str
    related_rental_id: int | None
    related_rental_code: str | None
    created_at: datetime

File: frontend/src/api/wallet.ts
import { apiRequest } from "./client";

export interface Wallet {
  balance_credits: number;
}

export interface WalletLedgerEntry {
  id: number;
  delta_credits: number;
  reason: string;
  related_rental_id: number | null;
  related_rental_code: string | null;
  created_at: string;
}

export const getWallet = (): Promise<Wallet> => apiRequest<Wallet>("/wallet/me");

export const getWalletLedger = (): Promise<WalletLedgerEntry[]> =>
  apiRequest<WalletLedgerEntry[]>("/wallet/ledger");
